src/code/z_play.c

```diff
@@ -1,81 +1,1608 @@
+                           s32 pixelBottom, s32 type) {
+    s32 i;
+    s32 j;
+    u32 pixel;
+    u32 r;
+    u32 g;
+    u32 b;
+
+    switch (type) {
+        case 4: {
+            u8* destI4 = (u8*)destI;
+            u32 upper;
+            u32 lower;
+
+            for (i = pixelTop; i <= pixelBottom; i++) {
+                for (j = pixelLeft; j <= pixelRight; j += 2) {
+                    pixel = srcBuf[i * bufWidth + j];
+                    r = (pixel >> 11) & 0x1F;
+                    g = (pixel >> 6) & 0x1F;
+                    b = (pixel >> 1) & 0x1F;
+                    upper = ((r * 2 + g * 4 + b) * 0xF) / 217;
+
+                    pixel = srcBuf[i * bufWidth + j + 1];
+                    r = (pixel >> 11) & 0x1F;
+                    g = (pixel >> 6) & 0x1F;
+                    b = (pixel >> 1) & 0x1F;
+                    lower = ((r * 2 + g * 4 + b) * 0xF) / 217;
+
+                    *(destI4++) = (upper << 4) | lower;
+                }
+            }
+            break;
+        }
+
+        case 5: {
+            u8* destI5 = (u8*)destI;
+
+            for (i = pixelTop; i <= pixelBottom; i++) {
+                for (j = pixelLeft; j <= pixelRight; j++) {
+                    pixel = srcBuf[i * bufWidth + j];
+                    r = (pixel >> 11) & 0x1F;
+                    g = (pixel >> 6) & 0x1F;
+                    b = (pixel >> 1) & 0x1F;
+
+                    pixel = 0;
+
+                    *(destI5++) = (((r * 2 + g * 4 + b) * 0xFF) / 217) & 0xF8;
+                }
+            }
+            break;
+        }
+
+        case 8: {
+            u8* destI8 = (u8*)destI;
+
+            for (i = pixelTop; i <= pixelBottom; i++) {
+                for (j = pixelLeft; j <= pixelRight; j++) {
+                    pixel = srcBuf[i * bufWidth + j];
+
+                    r = (pixel >> 11) & 0x1F;
+                    g = (pixel >> 6) & 0x1F;
+                    b = (pixel >> 1) & 0x1F;
+
+                    *(destI8++) = ((r * 2 + g * 4 + b) * 0xFF) / 217;
+                }
+            }
+            break;
+        }
+
+        case 16: {
+            u16* destI16 = (u16*)destI;
+
+            for (i = pixelTop; i <= pixelBottom; i++) {
+                for (j = pixelLeft; j <= pixelRight; j++) {
+                    *(destI16++) = srcBuf[i * bufWidth + j];
+                }
+            }
+            break;
+        }
+    }
+}
+
+void Play_SetMotionBlurPriorityAlpha(u32 alpha) {
+    R_MOTION_BLUR_PRIORITY_ALPHA = alpha;
+}
+
+void Play_EnableMotionBlurPriority(u32 alpha) {
+    R_MOTION_BLUR_PRIORITY_ALPHA = alpha;
+    R_MOTION_BLUR_PRIORITY_ENABLED = true;
+}
+
+void Play_DisableMotionBlurPriority(void) {
+    R_MOTION_BLUR_PRIORITY_ENABLED = false;
+}
+
+typedef enum {
+    /* 0 */ PICTOGRAPH_PHOTO_STATE_OFF,
+    /* 1 */ PICTOGRAPH_PHOTO_STATE_START,
+    /* 2 */ PICTOGRAPH_PHOTO_STATE_TAKE,
+    /* 3 */ PICTOGRAPH_PHOTO_STATE_FINISH
+} PictographPhotoState;
```

Comment:
Because `R_PICTOGRAPH_PHOTO_STATE` also appears in `z_parameter.c`, could you move this to `z64.h`? I also introduce a `z64interface.h` in my parameter branch, so maybe it could end up there? Either way, it's not local to this file